server/route.go

```diff
@@ -1011,7 +1032,15 @@ func (c *client) sendRouteSubOrUnSubProtos(subs []*subscription, isSubProto, tra
 			// the lock, which could cause pingTimer to think that this
 			// connection is stale otherwise.
 			c.last = time.Now()
-			c.flushOutbound()
+			if !c.flushOutbound() {
+				// It means that we could not flush because another go
+				// routine (likely the writeLoop) is doing (or is done with)
+				// the socket write and is waiting for the lock. Give it
+				// a chance to complete its operation by releasing
+				// the lock here.
+				c.mu.Unlock()
```

Comment:
This makes sense but looks weird..

Reply:
Totally agree and I think we do that in another place. We should not put the onus on the caller of flushOutbound() to do that every single time. It may make more sense to do it inside flushOutbound() when returning false.

Reply:
Any better approaches in your mind?

Reply:
Maybe.. I have realized something: why do we apply max payload to non client connections? I am not sure this was the intent. If we don't, then I guess it is ok if a routine under the lock keeps calling queueOutbound() since there would be no risk to create a slow consumer due to max pending bytes. We could still call flushOutbound() from time to time, but would not have to worry about the scenario described before (say writeLoop not able to grab lock but still have flush flag set).
I see that for routes and internal client we create the connection with `mpay: -1` (we don't do that for gateways and leaf nodes btw), but problem is that initClient() initializes c.out.mp to the option's MaxPayload. We could that only for c.kind == CLIENT maybe. Obviously, if we want to ignore c.out.pb for non clients, we need to change queueOutbound() to check getting over c.out.mp only if c.out.mp > 0.

Reply:
We apply it to all connections but mostly is inherited from config. Users under operator mode pick it up from account JWT.

Reply:
Ah I see now what you are saying.